temporaltest/server.go

```diff
@@ -39,10 +40,43 @@ func (ts *TestServer) fatal(err error) {
 }
 
 // Worker registers and starts a Temporal worker on the specified task queue.
+// WorkflowPanicPolicy is set to worker.FailWorkflow
 func (ts *TestServer) Worker(taskQueue string, registerFunc func(registry worker.Registry)) worker.Worker {
-	w := worker.New(ts.Client(), taskQueue, worker.Options{
-		WorkflowPanicPolicy: worker.FailWorkflow,
-	})
+	ts.defaultWorkerOptions.WorkflowPanicPolicy = worker.FailWorkflow
+
+	w := worker.New(ts.Client(), taskQueue, ts.defaultWorkerOptions)
+	registerFunc(w)
+	ts.workers = append(ts.workers, w)
+
+	if err := w.Start(); err != nil {
+		ts.fatal(err)
+	}
+
+	return w
+}
+
+// NewWorkerWithOptions returns a Temporal worker on the specified task queue.
+// WorkflowPanicPolicy is set to worker.FailWorkflow
+func (ts *TestServer) NewWorkerWithOptions(taskQueue string, registerFunc func(registry worker.Registry), opts worker.Options) worker.Worker {
+	opts.WorkflowPanicPolicy = worker.FailWorkflow
+
+	w := worker.New(ts.Client(), taskQueue, opts)
+	registerFunc(w)
+	ts.workers = append(ts.workers, w)
+
+	if err := w.Start(); err != nil {
+		ts.fatal(err)
+	}
+
+	return w
+}
+
+// NewWorkerWithClient returns a Temporal worker with a custom client on the specified task queue.
+// WorkflowPanicPolicy is set to worker.FailWorkflow
+func (ts *TestServer) NewWorkerWithClient(client client.Client, taskQueue string, registerFunc func(registry worker.Registry), opts worker.Options) worker.Worker {
+	opts.WorkflowPanicPolicy = worker.FailWorkflow
+
+	w := worker.New(client, taskQueue, opts)
```

Comment:
🤔 I'm a little worried about having 3 different ways of constructing workers. I know this might be useful in some cases, but power users are already able to construct a new worker "by hand" if they need to, eg.

```go
func TestFoo(t *testing.T) {
	// Create test Temporal server and client
	ts := temporaltest.NewServer(temporaltest.WithT(t))
	c := ts.NewClientWithOptions(client.Options{
		// ...
	})

	// Register a new worker
	w := worker.New(c, "test", worker.Options{})
	defer w.Stop()
	// Register some workflows/activities
	// ...
	if err := w.Start(); err != nil {
		t.Fatal(err)
	}
}
```

It's actually kind of hard to come up with an example use case for this method that _couldn't_ be handled via a combination of `WithBaseClientOptions` and `NewWorkerWithOptions`.

So I'm thinking in the interest of keeping things simple for _most_ users we skip adding this method at least for now.